core/src/main/java/org/elasticsearch/plugins/RemovePluginCommand.java

```diff
@@ -56,8 +57,19 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
         execute(terminal, arg, env);
     }
 
-    // pkg private for testing
-    void execute(Terminal terminal, String pluginName, Environment env) throws Exception {
+/**
+    * Remove the plugin named pluginName.
+    *
+    * @param terminal       the Terminal to use for input/output
+    * @param pluginName     the name of the plugin to remove
+    * @param env            the environment for the local node
+    *
+    * @throws IOException   if the plugin is no longer available to delete or permission is denied
```

Comment:
I think we can state this more generally as it's thrown if any I/O exception occurs while performing a file operation (of which there are a few).